test.py

```diff
@@ -333,7 +334,16 @@ def __init__(self, path, cfg: dict, options: argparse.Namespace, mode: str) -> N
 
         self.create_cluster = self.topology_for_class(topology["class"], topology)
 
-        self.clusters = Pool(cfg.get("pool_size", 2), self.create_cluster)
+        cfg_pool_size = cfg.get("pool_size", 2)
+        # Max 1 scylla server per core
+        cores = psutil.cpu_count(logical=False)
+        rf = topology["replication_factor"]
+        rf_core = int(cores / rf)
+        assert rf_core >= 1, f"Not enough cores to run test with replication factor {rf}"
```

Comment:
I don't understand why you need to add this assert - it's obviously not fun to run a test with RF=3 on a 2-core antique, but why should we forbid it?
When we have few cores, we should limit unnecessary parallelism (i.e., pool_size), but not forbid necessary parallelism inherent in the test itself.

Reply:
Fixed in https://github.com/scylladb/scylladb/commit/146bbd7167d5d67890d32d33c35a440b343f4c02